yt/visualization/plot_modifications.py

```diff
@@ -105,7 +105,7 @@ def project_coords(self, plot, coord):
         code_length units.  Projected data units are 2D versions of the
         simulation data units relative to the axes of the final plot.
         """
-        if len(coord) == 3:
+        if coord.shape[0] == 3:
             if not isinstance(coord, YTArray):
                 coord = plot.data.ds.arr(coord, 'code_length')
```

Comment:
Given the change to pass coord through `np.array` is this block of the if statement ever executed?

Reply:
Good catch.  This actually helped me catch a corner case because I incorrectly assumed that when you wrap a YTArray with a numpy array, that it retains its YTArray-ness.  It does not.  So now I make a check at the beginning of `sanitize_coord_system()` to assure things are either a YTArray or numpy array, but not to just clobber the YTArray if it already exists.  Thanks for the note!